osu.Game/Screens/Ranking/ScorePanel.cs

```diff
@@ -252,7 +262,14 @@ private void updateState()
 
                     bool firstLoad = topLayerContent == null;
                     topLayerContentContainer.Add(topLayerContent = new ExpandedPanelTopContent(Score.User, firstLoad) { Alpha = 0 });
-                    middleLayerContentContainer.Add(middleLayerContent = new ExpandedPanelMiddleContent(Score, displayWithFlair) { Alpha = 0 });
+
+                    if (isNewLocalScore && expandedPanelMiddleContent != null)
+                        expandedPanelMiddleContent?.Show();
+                    else
+                    {
+                        middleLayerContentContainer.Add(expandedPanelMiddleContent = middleLayerContent = new ExpandedPanelMiddleContent(Score, displayWithFlair) { Alpha = 0 });
+                        if (expandedPanelMiddleContent != null) expandedPanelMiddleContent.AlwaysPresent = true;
```

Comment:
I don't get it. The `AlwaysPresent` set is presumably the fix here yes? Then what is going on with _everything else_ in this diff?

Why does `isNewLocalScore` exist? What's with the random changes to hiding and showing the panel?

Reply:
In the original implementation, the `top`/`middleLayerContent` would expire upon changing the state. My implementation prevents `middleLayerContent` from being deleted if the `ScorePanel`'s `displayWithFlair` is true, thus preventing the sound from stopping.

I removed `isNewLocalScore` and replaced all usages with `displayWithFlair`. Consequently, I also removed the line of code that was setting `displayWithFlair` since it is no longer used (it was previously used in constructing `ExpandedPanelMiddleContent`, which is now only constructed once and then shown/hidden).

Additionally, I added `AlwaysPresent` to the `expandedPanelMiddleContent`. Without this, when `Alpha` was 0, the sound would stop.